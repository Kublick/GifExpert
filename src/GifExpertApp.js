import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
	//	const categories = ["One Punch", "Samurai X", "Dragon Ball"];

	const [categories, setCategories] = useState(defaultCategories);

	const handleAdd = () => {
		// setCategories([...categories, "HunterXHunter"]);
	};

	return (
		<div>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />
			<ul>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ul>
		</div>
	);
};
