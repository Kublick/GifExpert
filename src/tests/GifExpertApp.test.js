import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas GifExpertApp", () => {
	test("Debe ser igual que el snapshot", () => {
		const wrapper = shallow(<GifExpertApp />);

		expect(wrapper).toMatchSnapshot();
	});

	test("debe de mostrar una lista de categorias", () => {
		const categories = ["One Punch", "Dragon Ball"];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

		console.log("que", wrapper.find("GifGrid").length);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("GifGrid").length).toBe(categories.length);
	});
});
