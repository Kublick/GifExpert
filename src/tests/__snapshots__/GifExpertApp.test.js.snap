// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp Debe ser igual que el snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</div>
`;

exports[`Pruebas GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ul>
</div>
`;
